modules/mono/editor/hostfxr_resolver.cpp

```diff
@@ -0,0 +1,305 @@
+	// When emulating x64 on arm
+	String dotnet_root_emulated = path::join(r_dotnet_root, "x64");
+	if (FileAccess::exists(path::join(dotnet_root_emulated, "dotnet"))) {
+		r_dotnet_root = dotnet_root_emulated;
+		return true;
+	}
+#endif
+
+	return true;
+#else
+	r_dotnet_root = "/usr/share/dotnet";
+	return true;
+#endif
+}
+
+bool get_install_location_from_file(const String &p_file_path, String &r_dotnet_root) {
+	Error err = OK;
+	Ref<FileAccess> f = FileAccess::open(p_file_path, FileAccess::READ, &err);
+
+	if (f.is_null() || err != OK) {
+		return false;
+	}
+
+	String line = f->get_line();
+
+	if (line.is_empty()) {
+		return false;
+	}
+
+	r_dotnet_root = line;
+	return true;
+}
+
+bool get_dotnet_self_registered_dir(String &r_dotnet_root) {
+#if defined(WINDOWS_ENABLED)
+	String sub_key = "SOFTWARE\\dotnet\\Setup\\InstalledVersions\\" + get_dotnet_arch();
+	Char16String value = String("InstallLocation").utf16();
+
+	HKEY hkey = NULL;
+	LSTATUS result = RegOpenKeyExW(HKEY_LOCAL_MACHINE, (LPCWSTR)(sub_key.utf16().get_data()), 0, KEY_READ | KEY_WOW64_32KEY, &hkey);
+	if (result != ERROR_SUCCESS) {
+		return false;
+	}
+
+	DWORD size = 0;
+	result = RegGetValueW(hkey, nullptr, (LPCWSTR)(value.get_data()), RRF_RT_REG_SZ, nullptr, nullptr, &size);
+	if (result != ERROR_SUCCESS || size == 0) {
+		RegCloseKey(hkey);
+		return false;
+	}
+
+	Vector<WCHAR> buffer;
+	buffer.resize(size / sizeof(WCHAR));
+	result = RegGetValueW(hkey, nullptr, (LPCWSTR)(value.get_data()), RRF_RT_REG_SZ, nullptr, (LPBYTE)buffer.ptrw(), &size);
+	if (result != ERROR_SUCCESS) {
+		RegCloseKey(hkey);
+		return false;
+	}
+
+	r_dotnet_root = String::utf16((const char16_t *)buffer.ptr());
+	RegCloseKey(hkey);
+	return true;
+#else
+	String install_location_file = path::join("/etc/dotnet", "install_location_" + get_dotnet_arch().to_lower());
+	if (get_install_location_from_file(install_location_file, r_dotnet_root)) {
+		return true;
+	}
+
+	if (FileAccess::exists(install_location_file)) {
+		// Don't try with the legacy location, this will fall back to the hard-coded default install location
+		return false;
+	}
+
+	String legacy_install_location_file = path::join("/etc/dotnet", "install_location");
+	return get_install_location_from_file(legacy_install_location_file, r_dotnet_root);
+#endif
+}
+
+bool get_file_path_from_env(const String &p_env_key, String &r_dotnet_root) {
+	String env_value = OS::get_singleton()->get_environment(p_env_key);
+
+	if (!env_value.is_empty()) {
+		env_value = path::realpath(env_value);
+
+		if (DirAccess::exists(env_value)) {
+			r_dotnet_root = env_value;
+			return true;
+		}
+	}
+
+	return false;
+}
+
+bool get_dotnet_root_from_env(String &r_dotnet_root) {
+	String dotnet_root_env = "DOTNET_ROOT";
+	String arch_for_env = get_dotnet_arch();
+
+	if (!arch_for_env.is_empty()) {
+		// DOTNET_ROOT_<arch>
+		if (get_file_path_from_env(dotnet_root_env + "_" + arch_for_env, r_dotnet_root)) {
```

Comment:
According to https://github.com/dotnet/designs/blob/main/accepted/2021/install-location-per-architecture.md the environment variable should be all upper case.

```suggestion
		if (get_file_path_from_env(dotnet_root_env + "_" + arch_for_env.to_upper(), r_dotnet_root)) {
```